assets/js/modules/adsense/datastore/service.js

```diff
@@ -0,0 +1,170 @@
+const { createRegistrySelector } = Data;
+
+export const selectors = {
+	/**
+	 * Gets a URL to the service.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state        Data store's state.
+	 * @param {Object} [args]       Object containing optional path and query args
+	 * @param {string} [args.path]  A path to append to the base url.
+	 * @param {Object} [args.query] Object of query params to be added to the URL.
+	 * @return {string} The URL to the service
+	 */
+	getServiceURL: createRegistrySelector( ( select ) => ( state, { path, query } = {} ) => {
+		const userEmail = select( CORE_USER ).getEmail();
+		if ( userEmail === undefined ) {
+			return undefined;
+		}
+
+		const baseURI = 'https://www.google.com/adsense/new';
+		const queryParams = query ? { ...query, authuser: userEmail } : { authuser: userEmail };
+		if ( path ) {
+			const sanitizedPath = `/${ path.replace( /^\//, '' ) }`;
+			return addQueryArgs( `${ baseURI }${ sanitizedPath }`, queryParams );
+		}
+		return addQueryArgs( baseURI, queryParams );
+	} ),
+
+	/**
+	 * Returns the URL for creating a new AdSense account.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {string} AdSense URL to create a new account.
+	 */
+	getCreateAccountURL: createRegistrySelector( ( select ) => () => {
+		const query = {
+			source: 'site-kit',
+			utm_source: 'site-kit',
+			utm_medium: 'wordpress_signup',
+		};
+		const siteURL = select( CORE_SITE ).getReferenceSiteURL();
+		if ( undefined !== siteURL ) {
+			query.url = siteURL;
+		}
+		return addQueryArgs( 'https://www.google.com/adsense/signup/new', query );
+	} ),
+
+	/**
+	 * Returns the URL to an AdSense account's overview page.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {string} AdSense account overview URL.
+	 */
+	getAccountURL: createRegistrySelector( ( select ) => () => {
+		const accountID = select( STORE_NAME ).getAccountID();
+
+		if ( accountID === undefined ) {
+			return undefined;
+		}
+		return select( STORE_NAME ).getServiceURL( { path: `${ accountID }/home`, query: { source: 'site-kit' } } );
+	} ),
+
+	/**
+	 * Returns the URL to an AdSense account's site overview page.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {string} AdSense account site overview URL.
+	 */
+	getAccountSiteURL: createRegistrySelector( ( select ) => () => {
+		const accountID = select( STORE_NAME ).getAccountID();
+		const siteURL = select( CORE_SITE ).getReferenceSiteURL();
+
+		if ( accountID === undefined || siteURL === undefined ) {
+			return undefined;
+		}
+
+		const query = {
+			// TODO: Check which of these parameters are actually required.
+			source: 'site-kit',
+			url: parseDomain( siteURL ) || siteURL,
+		};
+		return select( STORE_NAME ).getServiceURL( { path: `${ accountID }/sites/my-sites`, query } );
+	} ),
+
+	/**
+	 * Returns the URL to the AdSense site list overview
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {string} AdSense account site list overview URL.
+	 */
+	getAccountSitesURL: createRegistrySelector( ( select ) => ( ) => {
```

Comment:
Based on my other comment, let's rename this selector to `getAccountManageSitesURL` and update its usages accordingly.